docs/contributing/index.rst

```diff
@@ -1,13 +1,135 @@
+
+For all commits, you must run the tests locally with ``tox`` before pushing to the central repository.
+
+Preperations for a new Python version
++++++++++++++++++++++++++++++++++++++
+
+RestrictedPython should be updated for each new version of Python.
+To do so:
+
+* Read the changelog (`What's new in Python`_).
+* Copy and adjust the new AST Grammar (found under: `Python 3 AST`_) to ``/docs/contributing/ast/python<version>.ast``.
+* Add a new file ``changes_from<old_version>to<new_version>.rst`` in the directory ``/docs/contributing/``.
+  If the changes are significant, especially if related to security, then add a description of the changes in that file.
+* Add those files to the ``toctree`` directive in ``index.rst``.
+* For each new **AST Node** or functionality:
+
+  * Add tests to ``/tests/``.
+  * Add a ``visit_<AST Node>`` to ``/src/RestrictedPython/transformer.py``.
+
+    If the new AST Node should be enabled by default, without any modification, please add a ``visit_<AST Node>`` method such as the following:
+
+    .. code-block:: python
+
+        def visit_<AST Node>(self, node):
+            """Allow `<AST Node>` expressions."""
+            return self.node_contents_visit(node)
+
+    If the new AST Node should be disabled by default, add a ``visit_<AST Node>`` method such as the following:
+
+    .. code-block:: python
+
+        def visit_<AST Node>(self, node):
+            """`<AST Node>` expression currently not allowed."""
+            self.not_allowed(node)
+
+    Please note, that for all AST Nodes without an explicit ``visit_<AST Node>`` method, a default applies which denies the usage of this expression and functionality.
+    As we try to be **as explicit as possible**, all language features should have a corresponding ``visit_<AST Node>``.
+    That follows the Zen of Python:
+
+    .. code-block:: pycon
+        :emphasize-lines: 5
+
+        >>> import this
+        The Zen of Python, by Tim Peters
+
+        Beautiful is better than ugly.
+        Explicit is better than implicit.
+        Simple is better than complex.
+        Complex is better than complicated.
+        Flat is better than nested.
+        Sparse is better than dense.
+        Readability counts.
+        Special cases aren't special enough to break the rules.
+        Although practicality beats purity.
+        Errors should never pass silently.
+        Unless explicitly silenced.
+        In the face of ambiguity, refuse the temptation to guess.
+        There should be one-- and preferably only one --obvious way to do it.
+        Although that way may not be obvious at first unless you're Dutch.
+        Now is better than never.
+        Although never is often better than *right* now.
+        If the implementation is hard to explain, it's a bad idea.
+        If the implementation is easy to explain, it may be a good idea.
+        Namespaces are one honking great idea -- let's do more of those!
```

Comment:
I strongly dislike this paragraph: We are having coded a policy denying everything which is not explicitly allowed. IMHO this is explicit enough. Requiring a handler function for each AST node type makes it more complex:
* There is the handler function, which does not add any value.
* It also requires a test providing a valid code example, to keep the coverage at 100 %.
* The test has to run only on new enough python version to prevent syntax errors.
* …

You are requesting a maintenance burden here which does not seem to be appropriate  in my eyes. It just makes the project more complicated not more secure.

I never saw the need to deny each and every AST node type we do not support: There are currently probably more than a dozen AST node types which are not allowed by default but have no handler function. – No, please do no go and try to find them as it is fine as it is.